serde_codegen_internals/src/attr.rs

```diff
@@ -746,6 +785,64 @@ fn get_ser_and_de<T, F>(cx: &Ctxt,
     Ok((ser_item.get(), de_item.get()))
 }
 
+type TypeConversions<T> = (Option<T>, Option<T>, Option<T>, Option<T>);
```

Comment:
Make this a struct instead of a tuple so neither caller nor get_from_into_as can mix them up. Generally it is a bad idea to have tuples with many (more than 2) identical element types.